src/client_side.cc

```diff
@@ -657,28 +656,26 @@ clientSetKeepaliveFlag(ClientHttpRequest * http)
 }
 
 /// checks body length of non-chunked requests
-static int
+static bool
 clientIsContentLengthValid(HttpRequest * r)
```

Comment:
Please consider making this an HttpRequest method since you are rewriting most of it, it only needs HttpRequest object to make the decision, and it feels out of place in client_side.cc (we deal with HTTP requests in other parts of Squid as well).

Reply:
done in fc31d38